ValkeyJSON.md

```diff
@@ -0,0 +1,214 @@
+JSON data interchange standard. With this feature, users can natively store, query, and modify JSON data structures in 
+Valkey using the popular [JSONPath query language](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html). 
+To help users migrate from Redis and RedisJSON, as well as capitalize on existing OSS RedisJSON client libraries, the module 
+is designed to be API-compatible and RDB-compatible with Redis Ltd.’s RedisJSON v2.
+
+## Motivation
+
+JSON format is a widely used data exchange format and simplifies the development of applications that store complex data 
+structures by providing powerful searching and filtering capabilities. However, [Valkey core](https://github.com/valkey-io/valkey) 
+does not have a native data type for JSON. Redis Ltd.‘s RedisJSON is a popular Redis module, but not under a true 
+open source license and hence cannot be distributed freely with Valkey. There’s a demand in the Valkey 
+community to have a JSON module that matches most of the features of RedisJSON and is as API-compatible as possible. 
+See the community discussions [here](https://github.com/orgs/valkey-io/discussions?discussions_q=is%3Aopen+JSON).
+
+## Design Considerations
+
+ValkeyJSON will introduce a new JSON data type for Valkey, and commands to insert, update, delete and query JSON data. 
+To help users migrate from Redis and RedisJSON, as well as capitalize on existing OSS RedisJSON client libraries, ValkeyJSON 
+aims to be a drop-in replacement of RedisJSON. Therefore, it is designed be API-compatible and RDB-compatible with 
+Redis Ltd.’s RedisJSON.
+
+### RDB Compatibility
+
+To help users migrate from Redis and RedisJSON, ValkeyJSON will support RDB compatibility with RedisJSON. ValkeyJSON will 
+be able to load RDBs generated from RedisJSON. Likewise, RedisJSON will be able to load RDBs generated from ValkeyJSON.
+
+### Choice of JSON Library
+
+We have evaluated 7 open source JSON libraries - BSON, ION, MessagePack, yyjson, cJSON, Serde JSON, and RapidJSON. BSON, 
+ION and MessagePack do not have write (insert/update/delete) API and therefore cannot be used for the project. Among the 
+rest, RapidJSON stands out as both memory efficient and providing efficient insert/update/delete API.
+
+### JSONPath Query Parser
+
+[JSONPath](https://www.ietf.org/archive/id/draft-goessner-dispatch-jsonpath-00.html) is a query language for JSON with 
+features similar to XPath for XML. JSONPath is used for selecting and extracting elements from a JSON document. It 
+supports advanced query capabilities such as wildcard, filter expressions, slices, union, recursive search, etc.
+Using JSONPath query language, users can natively store, query, and modify JSON data structures, either wholly or partially.
+
+![](./images/jsonpath_parser_design.png)
```

Comment:
Can we move this to be a table inline with the text instead of a png? If anyone wants to comment or modify it, it will be difficult.

Reply:
Will remove the diagram. Also, based on other feedbacks, will make the design description stay at high level.